app/graph/page.tsx

```diff
@@ -83,7 +83,12 @@ export default function Page() {
 
     return (
         <div className="h-full flex flex-col p-2 gap-y-2">
-            <Query className="border rounded-lg border-gray-300 p-2" onSubmit={runQuery} onQueryUpdate={(state) => { queryState.current = state }} />
+            <Query
+                className="border rounded-lg border-gray-300 p-2"
+                onSubmit={runQuery}
+                onQueryUpdate={(state) => { queryState.current = state }}
+                deleteGraph={() => setGraph(Graph.empty())}
```

Comment:
Please rename to onDeleteGraph.
```suggestion
                onDeleteGraph={() => setGraph(Graph.empty())}
```

Why do you need to empty the Graph and not just call  `setGraph(null)`